src/coreclr/jit/rationalize.cpp

```diff
@@ -718,6 +718,14 @@ Compiler::fgWalkResult Rationalizer::RewriteNode(GenTree** useEdge, Compiler::Ge
             }
             break;
 
+        case GT_GCPOLL:
+        {
+            // GCPOLL is essentially a no-op, we used it as a hint for fgCreateGCPoll
+            BlockRange().Delete(comp, m_block, node);
+            node = comp->gtNewNothingNode();
```

Comment:
```suggestion
            assert(use.IsDummyUse());
            BlockRange().Delete(comp, m_block, node);
            return Compiler::WALK_CONTINUE;
```

Deleting the node is ok only because nothing has a node like this as an operand, so let's assert that.
Creating a new node here is unnecessary (and wouldn't work as is because nothing threads it into the block)